test/integration/kubernetes.go

```diff
@@ -52,6 +55,15 @@ func newK8sEnvironment(t *testing.T) k8sEnvironment {
 	testNamespace, err := clientset.CoreV1().Namespaces().Create(context.Background(), &namespaceTemplate, metav1.CreateOptions{})
 	require.NoError(t, err)
 
+	// Wait for namespace to be active before proceeding
+	err = retryUntilTrue(defaultTimeout, defaultBackoff, func() bool {
+		namespaces, err := clientset.CoreV1().Namespaces().Get(context.Background(), testNamespace.Name, metav1.GetOptions{})
+		require.NoError(t, err)
+
+		return namespaces.Status.Phase == "Active"
+	})
+	require.NoError(t, err)
+
```

Comment:
This logic follows the pattern established in the [addPodAndWaitOnPhase function](https://github.com/newrelic/newrelic-prometheus-configurator/blob/main/test/integration/kubernetes.go#L78-L95), so I would suggest that for the sake of code reusability you create a function in this file and instead call the function here.